### PR TITLE
Pill 20: Init according 19's teaser

### PR DESCRIPTION
**Pull request by Ericson2314**
WIP

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { pkgs ? import <nixpkgs> {}, revCount, shortRev }:
 let
   lib = pkgs.lib;
-  sources = lib.sourceFilesBySuffices ./. [ ".xml" ".txt" ];
+  sources = lib.sourceFilesBySuffices ./. [ ".xml" ".txt" ".bash" ];
```
Comment (Ericson2314) [inline default.nix]: I wanted nice syntax highlighting in my snippets.

**Comment by Ericson2314** [comment]
I suppose I should test that one can actually build the example derivations with the really old nixpkgs I am using for teaching purposes.

File: default.nix (hunk 1 above)
Comment (Ericson2314) [reply default.nix]: I'd also like to use `.nix`, but we should still filter out the top-level Nix files.

File: pills/20-basic-dependencies-and-hooks.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      Allowing dependencies to learn about their <emphasis>sibling</emphasis> dependencies is exactly what compiler need.
+    </para>
+
+  </section>
+
+  <section>
+    <title>Next pill...</title>
+    <para>
+      ...I'm not sure!
+      We could talk the additional dependency types and hooks which cross compilation necessitates, building on our knowledge here to cover stdenv as it works today.
```
Comment (Mic92) [inline pills/20-basic-dependencies-and-hooks.xml]: I propose you for a writeup on cross-compilation: @Ericson2314

Comment (Ericson2314) [reply pills/20-basic-dependencies-and-hooks.xml]: Hmm. Implementation wise, cross compilation is intentionally part of everything else, and not one distinct add-on. I suppose ideally I'd document each thing with cross compilation support and the overall methodology.

**Comment by dtzWill** [comment]
nixos-weekly? :D

**Pull request merged**